docs/Contributing.asciidoc

```diff
@@ -183,14 +187,68 @@ package, one of the tests consists of a call to
 http://perltidy.sourceforge.net/[Perltidy] to ensure that the contributed code
 follows the most common Perl style conventions.
 
-== Starting the webserver from local Git checkout
-* To start the webserver for development, use the +scripts/openqa daemon+.
-* The other daemons (mentioned in
-  the link:images/architecture.svg[architecture diagram]) are started in the
-  same way, e.g. +script/openqa-scheduler daemon+.
-* openQA will pull the required assets on the first run.
-* openQA uses SASS. Under openSUSE, installing +rubygem(sass)+ should be
-  sufficient.
+[[development-setup]]
+== Development setup
+For developing openQA and os-autoinst itself it makes sense to checkout the
+<<Contributing.asciidoc#repo-urls,Git repositories>> and to start the daemons manually.
+
+This section should give you a general idea how to do that. For a concrete example some developers
+use under openSUSE Tumbleweed have a look at the https://github.com/Martchus/openQA-helper[openQA-helper repository].
+
+=== Dependencies
+Have a look at the packaged version (e.g. +openQA.spec+ within the root of the openQA repository)
+for all required dependencies. For development build time dependencies need to be installed as well.
+Recommended dependencies such logrotate can be ignored. For openSUSE there is also the +openQA-devel+
+meta-package which pulls all required dependencies for development.
+
+[[setup-postgresql]]
+=== Setting up the PostgreSQL database
+One also needs to setup a PostgreSQL database for openQA manually owned by your regular user:
+
+1. Install PostgreSQL - under openSUSE the following package are required:
+   `postgresql-server postgresql-init`
+2. Start the server: `systemctl start postgresql`
+3. The next two steps need to be done as the user postgres: `su - postgres`
+4. Create user: `createuser your_username` where `your_username` must be the same
+   as the UNIX user you start your local openQA instance with.
+5. Create database: `createdb -O your_username openqa-local` where `openqa-local` is
+   the name you want to use for the database
+6. Configure openQA to use PostgreSQL as described in the section
+   <<Installing.asciidoc#database,Database>> of the installation guide.
+   User name and password are not required.
+7. openQA will default-initialize the new database on the next startup.
+
+==== Importing production data
```

Comment:
This should probably be marked as optional and mention that the developer is responsible for finding a production snapshot to use.